modules/platforms/cpp/client/src/cluster_connection.cpp

```diff
@@ -0,0 +1,321 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iterator>
+
+#include "common/utils.h"
+
+#include "ignite/network/codec.h"
+#include "ignite/network/codec_data_filter.h"
+#include "ignite/network/length_prefix_codec.h"
+#include "ignite/network/network.h"
+
+#include "ignite/protocol/writer.h"
+#include "ignite/protocol/reader.h"
+#include "ignite/protocol/utils.h"
+
+#include "cluster_connection.h"
+
+namespace ignite::detail
+{
+
+ClusterConnection::ClusterConnection(IgniteClientConfiguration configuration) :
+    m_configuration(std::move(configuration)),
+    m_pool(),
+    m_logger(m_configuration.getLogger()),
+    m_generator(std::random_device()()) { }
+
+void ClusterConnection::start(std::function<void(IgniteResult<void>)> callback)
```

Comment:
```suggestion
void ClusterConnection::startAsync(std::function<void(IgniteResult<void>)> callback)
```

Do we add `async` suffix to async methods?

Reply:
I guess, you are right.